### PR TITLE
VZ-4527 Troubleshooting Opensearch/Elasticsearch

### PR DESCRIPTION
**Pull request by bryantidd**

### CODE REVIEW

File: content/en/docs/troubleshooting/troubleshooting-elasticsearch.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+creationTimestamp:
+generation:
+resourceVersion:
+selfLink:
+uid:
+status:
+```
+and every line below status:
+
+edit the section like below to increase the storage to the same value as you did in the Verrazzano Operator:
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-elasticsearch.md:90]: edit  -> Edit
like below -> I would suggest removing these words -> suggested re-write:
Edit the section to increase the storage to the same value that you used for the Verrazzano operator

File: content/en/docs/troubleshooting/troubleshooting-elasticsearch.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+```
+storage: 200Gi
+```
+Save that file.
+
+The following command will delete the Statefulset, but allow the associated pods to continue to run.
+```
+$ kubectl -n verrazzano-system delete sts vmi-system-es-master --cascade=orphan
+```
+then run this command will re-create the Statefulset with the new volume size defined:
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-elasticsearch.md:100]: then -> Then
command will re-create -> command to recreate

File: content/en/docs/troubleshooting/troubleshooting-elasticsearch.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+```
+$ kubectl -n verrazzano-system apply -f vmi-system-es-master.yaml
+```
+The next steps are to one at a time, delete the existing master node pods, allowing the cluster to become healthy before moving on to the next node:
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-elasticsearch.md:104]: Suggested re-write:
The next steps are to delete the existing master node pods, one at a time,

File: content/en/docs/troubleshooting/troubleshooting-elasticsearch.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+```
+$ kubectl -n verrazzano-system delete pod/vmi-system-es-master-0 pvc/elasticsearch-master-vmi-system-es-master-0
+```
+Wait for the new master node pod to become ready and check the health of the cluster:
+```
+$ kubectl -n verrazzano-system exec -it vmi-system-es-master-0 -- curl http://127.0.0.1:9200/_cat/health
+```
+and
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-elasticsearch.md:112]: and -> And

File: content/en/docs/troubleshooting/troubleshooting-elasticsearch.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+```
+When the cluster is healthy, continue to the next master node:
+```
+$ kubectl -n verrazzano-system delete pod/vmi-system-es-master-1 pvc/elasticsearch-master-vmi-system-es-master-1
+```
+Wait for the new master node pod to become ready and check the health of the cluster:
+```
+$ kubectl -n verrazzano-system exec -it vmi-system-es-master-0 -- curl http://127.0.0.1:9200/_cat/health
+```
+and
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-elasticsearch.md:124]: and -> And

File: content/en/docs/troubleshooting/troubleshooting-elasticsearch.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+```
+When the cluster is healthy, continue to the next master node:
+```
+$ kubectl -n verrazzano-system delete pod/vmi-system-es-master-2 pvc/elasticsearch-master-vmi-system-es-master-2
+```
+Wait for the new master node pod to become ready and check the health of the cluster:
+```
+$ kubectl -n verrazzano-system exec -it vmi-system-es-master-0 -- curl http://127.0.0.1:9200/_cat/health
+```
+and
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-elasticsearch.md:136]: and -> And

File: content/en/docs/troubleshooting/troubleshooting-elasticsearch.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+```
+$ kubectl -n verrazzano-system exec -it vmi-system-es-master-0 -- curl http://127.0.0.1:9200/_cat/indices
+```
+When the cluster is healthy rescale the operators:
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-elasticsearch.md:140]: healthy  -> healthy, (comma)

**Review by rosemarymarano — COMMENTED** [review]
Some edits and suggestions. Please let me know if anything is not clear.

**Comment by kminder** [comment]
The beginning might benefit from a bit more explanation.  In particular when the Verrazzano operator's ES_DATA_NODE_REPLICAS is changed to 3 that results in the creation of a third data node with PVs of the new size.  Then the instructions stop through fixing the old two data nodes.  There is never any mention of this new data node and that is does not need to be fixed.

**Comment by kminder** [comment]
The changes made by following this procedure will not survive upgrade right?  If so some warning should be added along with what can be done about it.

File: content/en/docs/troubleshooting/troubleshooting-elasticsearch.md [hunk 8]
```diff
@@ -1,145 +1,139 @@
...
-status:
-```
-and every line below status:
+1. First:
+   ```
+   $ kubectl -n verrazzano-system scale deploy verrazzano-operator --replicas=0
+   $ kubectl -n verrazzano-system scale deploy verrazzano-monitoring-operator --replicas=0
+   $ kubectl -n verrazzano-system get sts vmi-system-es-master -o yaml > vmi-system-es-master.yaml
+   ```
+2. Edit the file created in the previous command, vmi-system-es-master.yaml
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-elasticsearch.md:77]: vmi-system-es-master.yaml -> `vmi-system-es-master.yaml` (code font - My fault, I missed this one)

**Review by rosemarymarano — APPROVED** [review]
One minor thing I missed, otherwise ok.

File: content/en/docs/troubleshooting/troubleshooting-elasticsearch.md [hunk 9]
```diff
@@ -136,4 +136,28 @@ with a volume template in a StatefulSet, you must follow this procedure:
     $ kubectl -n verrazzano-system scale deploy verrazzano-operator --replicas=1
     $ kubectl -n verrazzano-system scale deploy verrazzano-monitoring-operator --replicas=1
     ```
+## Update the Verrazzano Custom Resource
+Now you will edit the Verrazzano CR, so that when you upgrade, the above changes will not be overwritten. Follow this procedure:
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-elasticsearch.md:140]: Now  -> Now, (comma)
the above changes -> the previous changes

File: content/en/docs/troubleshooting/troubleshooting-elasticsearch.md [hunk 10]
```diff
@@ -136,4 +136,28 @@ with a volume template in a StatefulSet, you must follow this procedure:
...
+
+1. Get the namespace and name of the Verrazzano CR
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-elasticsearch.md:142]: Please put a period at the end of each sentence.

**Review by rosemarymarano — COMMENTED** [review]
Minor edits.

**Review by rosemarymarano — APPROVED** [review]

**Pull request merged**